### DISCUSSION TITLE
What is the recommended way to create a CustomField for Joomla 5 with a custom/own FormField?

### DISCUSSION THREAD

**Discussion by astridx** [joomla/joomla-cms#42246] [Q&A] [answered]
What is the recommended way to create a CustomField for Joomla 5 with a custom/own FormField?

My [attempts so far](https://codeberg.org/astrid/j4examplecode/src/branch/t42/src/plugins/fields/agosmsmarker) have ended in code that does not load the form field. 

What am I doing wrong or how should I do this?

**Comment by Fedik** [comment, marked as answer]
For namespaced field you need to use `FormHelper::addFieldPrefix()` to register your namespace in the form, instead of `FormHelper::addFieldPath()`.
Example `FormHelper::addFieldPrefix('Foo\Bar')` for field `Foo\Bar\BeerField`
